design-documents/graph-ql/coverage/directory.graphqls

```diff
@@ -0,0 +1,48 @@
+# Copyright © Magento, Inc. All rights reserved.
+# See COPYING.txt for license details.
+
+type Query {
+    currency: Currency @resolver(class: "Magento\\DirectoryGraphQl\\Model\\Resolver\\Currency") @doc(description: "The currency query returns information about store currency.") @cache(cacheable: false)
+    countries: [Country] @resolver(class: "Magento\\DirectoryGraphQl\\Model\\Resolver\\Countries") @doc(description: "The countries query provides information for all countries. Sorted by Top Destination and locale specific country name.") @cache(cacheable: false)
+    country (id: String): Country @resolver(class: "Magento\\DirectoryGraphQl\\Model\\Resolver\\Country") @doc(description: "The countries query provides information for a single country.") @cache(cacheable: false)
+}
+
+type Currency {
+    base_currency_code: String
+    base_currency_symbol: String
+    default_display_currecy_code: String @deprecated(reason: "Symbol was missed. Use `default_display_currency_code`.")
+    default_display_currency_code: String
+    default_display_currecy_symbol: String @deprecated(reason: "Symbol was missed. Use `default_display_currency_symbol`.")
+    default_display_currency_symbol: String
+    available_currency_codes: [String]
+    exchange_rates: [ExchangeRate]
+}
+
+type ExchangeRate {
+    currency_to: String
+    rate: Float
+}
+
+type Country {
+    id: String
+    two_letter_abbreviation: String @doc(description: "Country code according to ISO 3166-1 alpha-2")
+    three_letter_abbreviation: String @doc(description: "Country code according to ISO 3166-1 alpha-3") 
+    name: String @doc(description: "Name of the country in the locale of the current store") 
+    full_name_locale: String @deprecated(reason: "Use `name`") 
+    full_name_english: String @deprecated(reason: "Use `name`")
+    region_required: Boolean @doc(description: "Is a the region for this country") 
+    postcode_required: Boolean @doc(description: "Is a postcode required for this country") 
+    eu: Boolean @doc(description: "Country is a member of the European Union") 
```

Comment:
instead of having a specific region for EU, wouldn't it be better to have a "Zone" with value as EU or APAC EMEA, etc
There are other zones besides EU

Reply:
In the backend of Magento there is a specific configuration field that allows you to check EU countries. But I agree that zone would be better, but that would require modifications on the other layers as well.